services/secondary-storage/server/src/main/java/org/apache/cloudstack/storage/resource/IpTablesHelper.java

```diff
@@ -0,0 +1,66 @@
+//
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+//
+
+package org.apache.cloudstack.storage.resource;
+
+import com.cloud.utils.script.Script;
+import org.apache.log4j.Logger;
+
+public class IpTablesHelper {
+    public static final Logger LOGGER = Logger.getLogger(IpTablesHelper.class);
+
+    public static final String OUTPUT_CHAIN = "OUTPUT";
+    public static final String INPUT_CHAIN = "INPUT";
+    public static final String INSERT = " -I ";
+    public static final String APPEND = " -A ";
+
+    public static boolean needsAdding(String chain, String rule) {
+        Script command = new Script("/bin/bash", LOGGER);
+        command.add("-c");
+        command.add("iptables -C " + chain + " " + rule);
+
+        String commandOutput = command.execute();
+        boolean needsAdding = (commandOutput != null && commandOutput.contains("iptables: Bad rule (does a matching rule exist in that chain?)."));
+        LOGGER.debug(String.format("Rule [%s], %s need adding to [%s] : %s",
+                rule,
+                needsAdding ? "does indeed" : "doesn't",
+                chain,
+                commandOutput
+        ));
+        return needsAdding;
+    }
+
+    public static String addConditionally(String chain, boolean insert, String rule, String errMsg) {
+        LOGGER.info(String.format("Adding rule [%s] to [%s] if required.", rule, chain));
+        if (needsAdding(chain, rule)) {
+            Script command = new Script("/bin/bash", LOGGER);
+            command.add("-c");
+            command.add("iptables" + (insert ? INSERT : APPEND) + chain + " " + rule);
+            String result = command.execute();
+            LOGGER.debug(String.format("Executed [%s] with result [%s]", command, result));
+            if (result != null) {
+                LOGGER.warn(errMsg + result);
+                return errMsg + result;
+            }
+        } else {
+            LOGGER.warn("Rule already defined in SVM: " + rule);
```

Comment:
```suggestion
            LOGGER.warn("Rule already defined in SSVM: " + rule);
```

Reply:
I had the idea that this would/could be more generic code. Is it important?

Reply:
thanks for the review @weizhouapache , I addressed the rest but hope to leave this as is, as the changes may be usefull for any SVM.

Reply:
> I had the idea that this would/could be more generic code. Is it important?

@DaanHoogland 
it is trivial.
ignore it then